### PR TITLE
F-65: enable resource locking

### LINKED ISSUE #65: allow to lock ressources
To prevent accidental operations like resource deletion, [OpenNebula allow to lock resources](https://docs.opennebula.io/5.12/operation/users_groups_management/chmod.html#locking-resources).

We could enable this in terraform via a new schema argument, and lock ressource at creation (or only via updates).
Then to destroy a locked ressource this should done in two steps:
1. a first apply to unlock the ressource
2. destroy

### PR DESCRIPTION
**Pull request by treywelsh**
Enable locking for VM, VNet, Template resources.

Miss acceptance testing

### CODE REVIEW

**Comment by treywelsh** [comment]
I updated the PR to split the locking code in two parts.
 This allow to unlock or lock and update some resource elements in only one step without failure.

I also added an acceptance test on the image resource only.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## 0.4.0 (Unreleased)
+
+Features:
```
Comment (jaypif) [inline CHANGELOG.md]: Please use Capital letters here

File: opennebula/shared_schemas.go [hunk 2]
```diff
@@ -238,6 +239,26 @@ func tagsSchema() *schema.Schema {
 	}
 }
 
+var locktypes = []string{"USE", "MANAGE", "ADMIN", "ALL", "UNLOCK"}
+
+func lockSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		Computed:    true,
```
Comment (jaypif) [inline opennebula/shared_schemas.go]: maybe remove the computed here

Comment (treywelsh) [reply opennebula/shared_schemas.go]: Removed

**Review by jaypif — APPROVED** [review]

**Pull request merged**